Version 1.3.1 prohibits successing delta-requests

Version 1.3.1 removed a test to see if a request started with `http`.  This makes it harder to do successing deltapage requests.

I wrote a failing unittest, although I suspect there should allready be one, namely `Functional\DeltaQueryTest::testDeltaQuery()`.  I do not have access to groups, so I'm not able to test that one.

# Unittest
I added a patch for this test [as an attachment](https://github.com/microsoftgraph/msgraph-sdk-php/files/2256860/bugUnsuccesfullMailFoldersDeltasync.zip).

```php
// File: tests\Functional\MailTest.php
public function testMailGetFoldersWithDelta()
{
        $folderIterator = $this->_client
            ->createCollectionRequest('GET', '/me/mailFolders/delta')
            ->setReturnType(Model\MailFolder::class);

        while (!$folderIterator->isEnd())
        {
            $folders = $folderIterator->getPage();
        }

        $deltaLink = $folderIterator->getDeltaLink();

        $this->assertNotEmpty($deltaLink);

        $folderIterator = $this->_client
            ->createCollectionRequest('GET', $deltaLink)
            ->setReturnType(Model\MailFolder::class);

        while (!$folderIterator->isEnd()) {
            $folders = $folderIterator->getPage();
        }

        $deltaLink2 = $folderIterator->getDeltalink();
        $this->assertNotEquals($deltaLink,$deltaLink2);
}
```
# Test Result
```
E                                                                   1 / 1 (100%)

Time: 3.73 seconds, Memory: 6.00MB

There was 1 error:

1) MailTest::testMailGetFoldersWithDelta
GuzzleHttp\Exception\RequestException: cURL error 1: Protocol "v1.0https" not supported or disabled in libcurl (see http://curl.haxx.se/libcurl/c/libcurl-errors.html)

/home/int8/msgraph-sdk-php/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php:187
/home/int8/msgraph-sdk-php/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php:150
/home/int8/msgraph-sdk-php/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php:103
/home/int8/msgraph-sdk-php/vendor/guzzlehttp/guzzle/src/Handler/CurlHandler.php:43
/home/int8/msgraph-sdk-php/vendor/guzzlehttp/guzzle/src/Handler/Proxy.php:28
/home/int8/msgraph-sdk-php/vendor/guzzlehttp/guzzle/src/Handler/Proxy.php:51
/home/int8/msgraph-sdk-php/vendor/guzzlehttp/guzzle/src/PrepareBodyMiddleware.php:37
/home/int8/msgraph-sdk-php/vendor/guzzlehttp/guzzle/src/Middleware.php:30
/home/int8/msgraph-sdk-php/vendor/guzzlehttp/guzzle/src/RedirectMiddleware.php:70
/home/int8/msgraph-sdk-php/vendor/guzzlehttp/guzzle/src/Middleware.php:59
/home/int8/msgraph-sdk-php/vendor/guzzlehttp/guzzle/src/HandlerStack.php:67
/home/int8/msgraph-sdk-php/vendor/guzzlehttp/guzzle/src/Client.php:277
/home/int8/msgraph-sdk-php/vendor/guzzlehttp/guzzle/src/Client.php:125
/home/int8/msgraph-sdk-php/vendor/guzzlehttp/guzzle/src/Client.php:131
/home/int8/msgraph-sdk-php/src/Http/GraphRequest.php:257
/home/int8/msgraph-sdk-php/src/Http/GraphCollectionRequest.php:150
/home/int8/msgraph-sdk-php/tests/Functional/MailTest.php:146
/home/int8/msgraph-sdk-php/vendor/phpunit/phpunit/phpunit:52

ERRORS!
Tests: 1, Assertions: 1, Errors: 1.

Process finished with exit code 2
```
# Reason
`$folderIterator->getDeltaLink()` will result in a string like `https://graph.microsoft.com/v1.0/me/mailFolders/delta?$deltatoken={redactedDeltaToken}`.

Since there is no longer a check to see if an endpoint starts with http, the version will be prepended to the string, resulting in a faulty url like `v1.0https://graph.microsoft.com/v1.0/me/mailFolders/delta?$deltatoken={redactedDeltaToken}`.

--- a/src/Http/GraphRequest.php
+++ b/src/Http/GraphRequest.php
@@ -415,6 +415,11 @@ class GraphRequest
     */
     private function _getRequestUrl()
     {
+        //Send request with opaque URL
+        if (stripos($this->endpoint, "http") === 0) {
+            return $this->endpoint;
+        }
+
         return $this->apiVersion . $this->endpoint;
     }
 

